File: backend/src/main/java/bingol/campus/story/core/response/StoryDetails.java
package bingol.campus.story.core.response;

import bingol.campus.like.entity.Like;
import bingol.campus.post.core.response.CommentDetailsDTO;
import bingol.campus.post.core.response.LikeDetailsDTO;
import bingol.campus.student.core.response.SearchAccountDTO;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.List;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class StoryDetails {
    private UUID id;
    private String username; // Hikayeyi paylaşan kullanıcının adı
    private String photoUrl; // Hikayenin fotoğraf URL'si
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createdAt;
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime expiresAt;
    private boolean isActive;
    private long likeCount;
    private List<CommentDetailsDTO> comments;
    private List<SearchAccountDTO> viewing;
    private List<LikeDetailsDTO>likes;

}

File: frontend/lib/models/story_details.dart
import 'package:json_annotation/json_annotation.dart';
import 'comment_details_dto.dart';
import 'search_account_dto.dart';
import 'like_details_dto.dart';
part 'story_details.g.dart';

@JsonSerializable()
class StoryDetails {
  final String id; // UUID'ler String olarak temsil edilir
  final String username; // Hikayeyi paylaşan kullanıcının adı
  final String photoUrl; // Hikayenin fotoğraf URL'si
  @JsonKey(name: 'createdAt')
  final DateTime createdAt; // Oluşturulma tarihi
  @JsonKey(name: 'expiresAt')
  final DateTime expiresAt; // Son kullanma tarihi
  final bool isActive; // Aktif mi?
  final int likeCount; // Beğeni sayısı
  final List<CommentDetailsDTO> comments; // Yorumlar
  final List<SearchAccountDTO> viewing; // Görüntüleyenler
  final List<LikeDetailsDTO> likes; // Beğeniler

  StoryDetails({
    required this.id,
    required this.username,
    required this.photoUrl,
    required this.createdAt,
    required this.expiresAt,
    required this.isActive,
    required this.likeCount,
    required this.comments,
    required this.viewing,
    required this.likes,
  });

  factory StoryDetails.fromJson(Map<String, dynamic> json) => _$StoryDetailsFromJson(json);
  Map<String, dynamic> toJson() => _$StoryDetailsToJson(this);
} 